### ISSUE TITLE
Lightweight dependencies for presto-worldcereal in inference mode

### ISSUE THREAD

**Issue by kvantricht** [WorldCereal/presto-worldcereal#72]
Mostly directed to @gabrieltseng:

Currently, installing this package always drags in many dependencies relevant only for training, often even only for creating the original Presto training dataset using Google Earth Engine. This puts a lot of unnecessary weight on inference using openEO. I know one way would be to copy the relevant Presto stuff into the `worldcereal-classification` repository but in here we define UDFs that are being executed on the openEO backend. Therefore we cannot send specific stuff defined in `worldcereal-classification` itself. It's better to keep Presto WorldCereal datasets defined in this module, but to keep things lightweight, **can we relax dependencies for inference and having a separate set of dependencies only for training mode?**

Second question: is there a specific reason to have Python >= 3.9? OpenEO backend currently runs Python 3.8, which supports latest Pytorch, so we're wondering also if for inference Python 3.8 would suffice?
